### PR TITLE
scorpio broker helm charts added

### PR DESCRIPTION
**Pull request by arpitNEC**

### CODE REVIEW

File: charts/scorpio-broker-dist/templates/namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{{- if not (lookup "v1" "Namespace" .Values.namespace .Values.namespace ) }}
```
Comment (wistefan) [inline charts/scorpio-broker-dist/templates/namespace.yaml:1]: Why is this required? Namespaces are usually handled by helm.

File: charts/scorpio-broker-dist/values.yaml [hunk 2]
```diff
@@ -0,0 +1,702 @@
+## namespace for deployment
```
Comment (wistefan) [inline charts/scorpio-broker-dist/values.yaml:1]: We are trying to comply with the helm best practices. Can you please add some short documentation to all values? See https://helm.sh/docs/chart_best_practices/values/#document-valuesyaml
In order to get a nice markdown file for the folder documentation of your chart, check this tool: https://github.com/norwoodj/helm-docs
All other charts use it to generate their documentation(f.e. https://github.com/FIWARE/helm-charts/tree/main/charts/orion) and it would help to get a consistent presentation of the charts.

File: charts/scorpio-broker-dist/values.yaml [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+  name: eureka
+## Number of eureka instances to deploy
+##
+  replicas: 1
+## Scorpio Broker eureka image version
+## ref: https://hub.docker.com/r/scorpiobroker/scorpio
+## Please use only latest version of docker images as few older images might not work with kubernetes. 
+##
+  image:
+   repository: scorpiobroker/scorpio
```
Comment (wistefan) [inline charts/scorpio-broker-dist/values.yaml:16]: According to the Chart.yaml, this deploys version 1.16.0 as a default. All image tags are "latest". This will lead to unexpected versions being deployed. Please use fixed versions as a default, to avoid the components being silently updated.

**Review by wistefan — CHANGES_REQUESTED** [review]
Thank you for your contribution. It looks good as a starting point, but a couple of things still need to be done.
1. Please decide if both of them are required and if so, carefully document on when to use which one. Running all components inside one container is usually not a good idea in production scenarios, so that the usage of the aaio chart needs to be clarified.
2. A couple of helm best-practices should be applied. Please see my comments on the "atContextServer" templates. They apply to the other components in the dist-chart, too. As a summary:
> make namespace configurable/deactivateable, its usually handled by helm
> apply the helm naming conventions to the values
> generate the names for all components, by using a _helpers.tpl(as you already do in the aaio chart)
> eureka address is hard-coded and will fail in case of different service names. Needs to be taken from the values
> update-strategy should be configurable, so that it can work with different clusters and deployment scenarios
> liveness-probe should be added to handle runtime failures
> empty blocks(e.g. status: {}) should be removed
3. How are the services are configured? None of them uses Env-vars or offers a configmap(f.e. holding application.properties or application.yaml) to configure the applications. How can I f.e. connect them to databases and where should the db's run anyways? Since you are using the actuator endpoint, it would f.e. also be good to have the ability to configure it, in order to prevent actuator functionality available to non-allowed users.
4. Please provide a readme to each chart, generated with the helm-docs tool: https://github.com/norwoodj/helm-docs to ensure consistency between the charts in the repo.       

Best,
Stefan

File: charts/scorpio-broker-dist/templates/entity-manager-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+              command:
+              - /bin/sh
+              - -c
+              - curl -X PUT {{ .Values.entityManager.eurekaAddress }}
+        {{- if .Values.entityManager.readinessProbe.enabled }}
+        readinessProbe:
+          exec:
+            command:
+            - sh
+            - -c
```
Comment (wistefan) [inline charts/scorpio-broker-dist/templates/entity-manager-deployment.yaml:48]: Why dont you use an http check instead? It would allow you to get rid of curl inside the container.

File: charts/scorpio-broker-dist/templates/eureka-node-port.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+metadata:
+  labels:
+    service: {{ .Values.eureka.eurekaLabel }}
+  name: {{ include "eureka.fullname" . }}-node-port
+  namespace: {{ .Release.Namespace }}
+spec:
+  type: {{ .Values.eureka.service.type }}
+  ports:
+  - port: {{ .Values.eureka.service.port }}
+    targetPort: {{ .Values.eureka.service.port }}
```
Comment (wistefan) [inline charts/scorpio-broker-dist/templates/eureka-node-port.yaml:13]: The nodeport should only be enabled if the service is of type nodeport

File: charts/scorpio-broker-dist/templates/query-manager-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  {{- with .Values.queryManager.updateStrategy }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+  template:
+    metadata:
+      labels:
+        service: {{ include "query-manager.fullname" . }}
+    spec:
+      containers:
+      - image: {{ .Values.queryManager.image.repository }}:{{ .Values.queryManager.image.tag }}
```
Comment (wistefan) [inline charts/scorpio-broker-dist/templates/query-manager-deployment.yaml:25]: Why did you decide to use that config mechanism? Its a little uncomfortable, since a user then has to override all configurations, instead of only those needed. Why not configuer the spring-boot application directly through its environment config mechanism?

File: charts/scorpio-broker-dist/templates/scorpio-gateway-node-port-svc.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+{{- if .Values.gateway.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    service: {{ .Values.gateway.name }}
+  name: scorpio-gateway-node-port
+  namespace: {{ .Release.Namespace }}
+spec:
```
Comment (wistefan) [inline charts/scorpio-broker-dist/templates/scorpio-gateway-node-port-svc.yaml:9]: This should be configurable. It will fail in environments where nodeport biding is not allowed or 30001 is already used by someone else.

File: charts/scorpio-broker-dist/templates/subscription-manager-deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        readinessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            - curl --silent http://localhost:2025/actuator/health | grep --quiet -e
+              '^{\"status\"\:\"UP\".*}$'
+          failureThreshold: {{ .Values.subscriptionManager.readinessProbe.failureThreshold }}
+          initialDelaySeconds: {{ .Values.subscriptionManager.readinessProbe.initialDelaySeconds }}
+          periodSeconds: {{ .Values.subscriptionManager.readinessProbe.periodSeconds }}
```
Comment (wistefan) [inline charts/scorpio-broker-dist/templates/subscription-manager-deployment.yaml:53]: Indention is a little to deep

File: charts/scorpio-broker-dist/values.yaml [hunk 9]
```diff
@@ -0,0 +1,982 @@
+## namespace for deployment
+namespace: scorpio-broker
+
+################################### SpringArgs #######################################################################
```
Comment (wistefan) [inline charts/scorpio-broker-dist/values.yaml:4]: Are there no cases where it makes sense to use different arguments for the different services?

File: charts/scorpio-broker-dist/values.yaml [hunk 10]
```diff
@@ -0,0 +1,982 @@
+## namespace for deployment
+namespace: scorpio-broker
```
Comment (wistefan) [inline charts/scorpio-broker-dist/values.yaml:2]: Still, setting the namespace that way needs to be avoided. namespaces are handled by helm.

File: charts/scorpio-broker-dist/values.yaml [hunk 11]
```diff
@@ -0,0 +1,982 @@
...
+################################### SpringArgs #######################################################################
+springArgs:
+  overrideSpringArgs: false
+  value: >
+
+################################### Eureka Server #######################################################################
+eureka:
+## Enable the eureka deployment & service
+  enabled: true 
+  name: eureka
```
Comment (wistefan) [inline charts/scorpio-broker-dist/values.yaml:13]: Whats the meaning of custom label? It would be good to use a mechanism like this for enabling custom labels:
https://github.com/FIWARE/helm-charts/blob/main/charts/orion/templates/deployment.yaml#L5-L6

File: charts/scorpio-broker-dist/values.yaml [hunk 12]
```diff
@@ -0,0 +1,982 @@
...
+  # limits: 
+  #   cpu: 250m
+  #   memory: 1Gi
+  # requests: 
+  #   cpu: 250m
+  #   memory: 256Mi
+  
+## gateway containers'  readiness probes. Evaluated as a template.
+## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes
+##
```
Comment (wistefan) [inline charts/scorpio-broker-dist/values.yaml:157]: Thats not an address, its a path. Since this uses $HOSTNAME (which I suppose is an env var), where is this configured? The containers usually do not have a hostname.

**Review by wistefan — CHANGES_REQUESTED** [review]

**Comment by wistefan** [comment]
@arpitNEC 
Hi, do you still work on this PR? There is still a number of open request and a couple of new once. Can you please check and resolve them? Let me know when I can review again.
Best,
Stefan

File: charts/scorpio-broker-dist/templates/query-manager-deployment.yaml (hunk 6 above)
Comment (ScorpioBroker) [reply charts/scorpio-broker-dist/templates/query-manager-deployment.yaml:25]: Hi, 
with the spring_args argument you would still just provide the configs that you want to change but it's than just one set config var for the user to override with his complete config instead of going through different env vars for spring boot.

File: charts/scorpio-broker-dist/values.yaml (hunk 9 above)
Comment (ScorpioBroker) [reply charts/scorpio-broker-dist/values.yaml:4]: Yes there might be scenarios where it makes sense to change the individual configs but the most common scenario will be for all of them same. Because in general people will configure postgres, kafka and eureka here which should be the same for all microservices

File: charts/scorpio-broker-dist/values.yaml (hunk 12 above)
Comment (RupamNEC) [reply charts/scorpio-broker-dist/values.yaml:157]: Hi,
This $HOSTNAME is not configured in env var as it is resolving into the IP address of the pod itself.

File: charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+        ports:
+        - containerPort: 27015
+          protocol: TCP
+        {{- if .Values.AtContextServer.readinessProbe.enabled }}
+        lifecycle:
+           preStop:
+            exec:
+              command:
+              - /bin/sh
+              - -c
```
[the remark this answers is not in the document]
Comment (RupamNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml:49]: Hi,
This preStop command gets executed on graceful shutdown only.

File: charts/scorpio-broker-dist/templates/entity-manager-deployment.yaml (hunk 4 above)
Comment (RupamNEC) [reply charts/scorpio-broker-dist/templates/entity-manager-deployment.yaml]: > @arpitNEC Hi, do you still work on this PR? There is still a number of open request and a couple of new once. Can you please check and resolve them? Let me know when I can review again. Best, Stefan

Hi @wistefan 
We have incorporated the review comments on this PR. Please review it again.
BR,
@RupamNEC

**Comment by ScorpioBroker** [comment]
Hi @wistefan, 

we addressed the review comments. I think there was a communication hiccup. We would also provide one more updated version within the next week because we some updates on the scorpio side we need to address as well.

BR
Benjamin

**Comment by RupamNEC** [comment]
Hi @wistefan,

we addressed the old review comments and updated the chart with a new version because of some updates from Scorpio. Kindly review it again. 

BR
Rupam

**Comment by martin-p-bauer** [comment]
@wistefan What is the status of this request - did you review again after the changes by Rupam?

**Comment by wistefan** [comment]
> @wistefan What is the status of this request - did you review again after the changes by Rupam?

Hi @martin-p-bauer ,
not yet, since there are a lot of unresolved conversations. Please comment and close them first.

File: charts/scorpio-broker-aaio/templates/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,59 @@
+apiVersion: apps/v1
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-aaio/templates/deployment.yaml:1]: We have added health/liveness probe. Please review again.

File: charts/scorpio-broker-aaio/templates/deployment.yaml [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+        {{- include "scorpioBroker-aaio.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "scorpioBroker-aaio.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-aaio/templates/deployment.yaml:30]: We have added env vars for kafka as well. Please review again

File: charts/scorpio-broker-aaio/values.yaml [hunk 16]
```diff
@@ -0,0 +1,79 @@
+# Default values for scorpioBroker-aaio.
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-aaio/values.yaml:1]: We have added short documentation accordingly.

File: charts/scorpio-broker-dist/Chart.yaml [hunk 17]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v2
+name: scorpioBroker
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/Chart.yaml]: We have added the description part to differentiate between aaio and dist. Please review again

File: charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml [hunk 18]
```diff
@@ -0,0 +1,59 @@
+{{- if .Values.AtContextServer.enabled }}
+apiVersion: apps/v1
+kind: Deployment
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml:3]: Updated files accordingly. Please review them

File: charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+metadata:
+  labels:
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml:6]: We have used helm naming convention. Please review again

File: charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+    service: {{ .Values.AtContextServer.name }}
+  name: {{ .Values.AtContextServer.name }}
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+  namespace: {{.Values.namespace}}
+spec:
+  replicas: {{ .Values.AtContextServer.replicas }}
+  selector:
+    matchLabels:
+      service: {{ .Values.AtContextServer.name }}
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml:19]: We have added _helpers.tpl file and incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+        service: {{ .Values.AtContextServer.name }}
+    spec:
+      containers:
+      - image: {{ .Values.AtContextServer.image.repository }}:{{ .Values.AtContextServer.image.tag }}
+        imagePullPolicy: {{ .Values.AtContextServer.image.pullPolicy }}
+        name: {{ .Values.AtContextServer.name }}
+        ports:
+        - containerPort: 27015
+          protocol: TCP
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml:42]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/atcontext-server-service.yaml [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+    service: {{ .Values.AtContextServer.name }}
+  name: {{ .Values.AtContextServer.name }}
+  namespace: {{.Values.namespace}}
+spec:
+  ports:
+  - name: "27015"
+    port: 27015
+    targetPort: 27015
+  selector:
+    service: {{ .Values.AtContextServer.name }}
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-service.yaml]: We have removed the empty blocks.

File: charts/scorpio-broker-dist/templates/eureka-node-port.yaml [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+metadata:
+  labels:
+    service: {{ .Values.eureka.name }}
+  name: {{ .Values.eureka.name }}-node-port
+  namespace: {{.Values.namespace}}
+spec:
+  type: NodePort
+  ports:
+  - port: 8761
+    targetPort: 8761
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/eureka-node-port.yaml]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/namespace.yaml (hunk 1 above)
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/namespace.yaml]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-aaio/templates/deployment.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-aaio/templates/deployment.yaml:30]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-aaio/values.yaml (hunk 16 above)
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-aaio/values.yaml:1]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/Chart.yaml (hunk 17 above)
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/Chart.yaml]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml (hunk 19 above)
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-deployment.yaml:6]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/atcontext-server-service.yaml (hunk 24 above)
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/atcontext-server-service.yaml]: Removed empty blocks from the files. Please review again

File: charts/scorpio-broker-dist/templates/eureka-node-port.yaml (hunk 25 above)
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/eureka-node-port.yaml]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/namespace.yaml (hunk 1 above)
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/namespace.yaml]: Removed namespace from the files, Please review again

File: charts/scorpio-broker-dist/values.yaml (hunk 2 above)
Comment (mohitNEC) [reply charts/scorpio-broker-dist/values.yaml]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/values.yaml (hunk 3 above)
Comment (mohitNEC) [reply charts/scorpio-broker-dist/values.yaml:40]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-aaio/templates/deployment.yaml (hunk 14 above)
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-aaio/templates/deployment.yaml:1]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/eureka-node-port.yaml (hunk 5 above)
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/eureka-node-port.yaml:12]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/values.yaml (hunk 11 above)
Comment (mohitNEC) [reply charts/scorpio-broker-dist/values.yaml:10]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/values.yaml (hunk 10 above)
Comment (mohitNEC) [reply charts/scorpio-broker-dist/values.yaml]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/values.yaml (hunk 9 above)
Comment (wistefan) [reply charts/scorpio-broker-dist/values.yaml:1]: But with the current solution, its not possible at all.

File: charts/scorpio-broker-dist/templates/scorpio-gateway-node-port-svc.yaml (hunk 7 above)
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/scorpio-gateway-node-port-svc.yaml:8]: We have incorporated the review comment. Please review again

File: charts/scorpio-broker-dist/templates/eureka-node-port.yaml [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+  labels:
+    {{- include "eureka.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.eureka.service.type }}
+  ports:
+  - port: {{ .Values.eureka.service.port }}
+    targetPort: {{ .Values.eureka.service.port }}
+    {{- if and (eq .Values.eureka.service.type "NodePort") .Values.eureka.service.nodePort }}
+    nodePort: {{ .Values.eureka.service.nodePort }}
+    {{- else if eq .Values.eureka.service.type "ClusterIP" }}
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-dist/templates/eureka-node-port.yaml:15]: Hi,
We have set it to nodeport so that our gateway can be accessed externally as well.

File: charts/scorpio-broker-aaio/templates/deployment.yaml [hunk 27]
```diff
@@ -0,0 +1,82 @@
...
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+    
+          env:
+            - name: POSTGRES_DB
+              value: {{ .Values.db.dbhost }}
+            - name: POSTGRES_PASSWORD
+              value: {{ .Values.db.password }}
+            - name: POSTGRES_USER
+              value: {{ .Values.db.user }}
```
[the remark this answers is not in the document]
Comment (mohitNEC) [reply charts/scorpio-broker-aaio/templates/deployment.yaml:60]: Hi,
Kafka is part of the image.

**Comment by amitrnec** [comment]
Hi @wistefan ,
We have incorporated all the comment please review so that it can be merge in main branch.

Thanks,
Amit Raghav.

**Review by wistefan — APPROVED** [review]
